source/builders.txt

```diff
@@ -0,0 +1,310 @@
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: aggregation, query, code example, type-safe
+
+Overview
+--------
+
+In this guide, you can learn about the **builder classes**
+that the {+library-short+} provides to create types used in your
+operations. You can use the builder classes and factory methods from the
+Aggregation Builder feature to create filters for other operations such
+as find, update, and delete operations. To learn more about the Aggregation
+Builder, see the :ref:`php-aggregation-builder-api` section of the
+Aggregation guide. 
+
+Using builders to create queries helps you identify errors at compile
+time and avoid them at runtime. This guide provides information on
+builder classes that you can use to perform the following tasks:
+
+- :ref:`php-builders-filter`
+- :ref:`php-builders-update`
+- :ref:`php-builders-changestream`
+
+.. note:: Setting Operation Options
+
+   You cannot specify options by using factory methods for the equivalent
+   aggregation stages. For example, you cannot use the ``Stage::limit()`` method
+   to set a returned documents limit on your find operation. You must specify
+   options by using the string-based syntax, as shown in the following code:
+   
+   .. code-block:: php
+   
+      $options = [
+          'limit' => 5,
+          '<option name>' => '<specification>',
+      ];
+
+This guide provides examples of how to use builders in non-aggregation
+operations. To view aggregation examples, see the :ref:`php-aggregation`
+guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``shipwrecks`` collection in the
+``sample_geospatial`` database from the :atlas:`Atlas sample datasets
+</sample-data>`. To access this collection from your PHP application,
+instantiate a ``MongoDB\Client`` that connects to an Atlas cluster
+and assign the following value to your ``$collection`` variable:
+
+.. literalinclude:: /includes/builders.php
+   :language: php
+   :dedent:
+   :start-after: start-db-coll
+   :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample
+datasets, see the :atlas:`Get Started with Atlas </getting-started>` guide.
+
+Import Builder Classes
+----------------------
+
+To run the examples in this guide, you must import the following classes
+into your application:
+
+.. literalinclude:: /includes/builders.php
+   :language: php
+   :dedent:
+   :start-after: start-imports
+   :end-before: end-imports
+
+.. _php-builders-filter:
+
+Create a Filter
+---------------
+
+You can use factory methods from the ``Query`` builder class to create
+filter definitions to use in find, update, and delete operations. To
+learn more about creating filters, see the :ref:`php-specify-query` guide.
+
+The following steps describe how to create a filter definition by using
+builders:
+
+1. Instantiate the ``QueryInterface`` by calling the ``Query::query()``
```

Comment:
I don't think we need to tell about the `QueryInterface`, that's an implementation detail.

The benefit of the `Query::query()` function is to create a query using **named argument syntax** and **type safety**.

`Query::query(foo: 'bar')` is equivalent to `['foo' => 'bar']`, but adds type contraint on the values.


For more advanced constraints using `$and` and `$or`, we don't need the "key". There is an example here: https://github.com/mongodb/mongo-php-library/blob/v1.x/tests/Builder/Query/AndOperatorTest.php

Reply:
I will add some explanation about the usefulness of using the Query class

Reply:
I will also change one of the examples to use and or or